### DISCUSSION TITLE
Recommendations for background processing

### DISCUSSION THREAD

**Discussion by colmexdev** [jeremyevans/roda#364] [General]
Hi there.

Recently I came across this amazing toolkit and decided to experiment with it by migrating a small app built in Rails. So far so good, but now I'm struggling with job processing.

I read the [conventions page](https://roda.jeremyevans.net/rdoc/files/doc/conventions_rdoc.html), but there's nothing about job processing. That's why I want to ask if there's a recommended way to handle this (in folder/code structure) and processors that can best fit.

Thank you in advance!

**Comment by jeremyevans** [comment]
Currently, Roda doesn't have conventions in that area.  If it isn't too much work, and you don't think you are currently benefiting much from the ActiveJob abstraction, I recommend switching to use the job processing API directly (e.g. Sidekiq).  Otherwise, if you want a simple transition, you could continue to use ActiveJob.

**Reply by colmexdev** [reply]
Then I'll go ahead and try what can best fit to the app in that case. Thank you very much @jeremyevans ! 😄

**Reply by jaredcwhite** [reply]
I'll give a second shoutout to Sidekiq, it takes just a bit of time to set up manually (aka outside of Rails) but otherwise is a very solid choice.

**Reply by colmexdev** [reply]
@jaredcwhite quite off-topic, but do you recommend Redis? The project I'm trying to migrate uses Beanstalkd, but it seems to be no longer maintained and also seems to have memory leaks every once in a while. Maybe it's a chance to also move to a better k/v store.

**Reply by jaredcwhite** [reply]
@colmexdev I am definitely no Redis expert, but I think it's basically become the workhorse of the industry at this point, much like PostgreSQL. You could also look into one of the new Redis forks, such as Valkey, but honestly if you're intending just to host something simple on Heroku or Render or whatever, it's probably not worth the time to investigate yet.

**Reply by adam12** [reply]
I'm using Beanstalkd via Backburner for Gemdocs.org. It works well enough. I'm surprised to hear about the memory leaks but I might just not be stressing it enough.

In many of my projects written using Roda (and Sequel, naturally) I've used Que to much success.
